docs/source/en/pipeline_tutorial.mdx

```diff
@@ -33,101 +33,295 @@ While each task has an associated [`pipeline`], it is simpler to use the general
+Now this is more appropriate !
+Really check out the hub, for different languages, models specialized in your field and more.
+And you can check out the live results directly from the hub to see if it fits or 
+handle corner cases better than another one.
+If you don't find any, start [`fine-tuning`]  # TODO FIX link.
+
+If you have several inputs, you can pass your input as a list:
 
 ```py
 >>> generator(
-...     "Three Rings for the Elven-kings under the sky, Seven for the Dwarf-lords in their halls of stone",
-...     num_return_sequences=2,
+...     [
+...         "https://huggingface.co/datasets/Narsil/asr_dummy/resolve/main/mlk.flac",
+...         "https://huggingface.co/datasets/Narsil/asr_dummy/resolve/main/1.flac"]
+...     ]
 ... )  # doctest: +SKIP
 ```
 
-### Choose a model and tokenizer
+If you want to iterate over a whole dataset, or want to use it for inference in a webserver, check out dedicated parts
 
-The [`pipeline`] accepts any model from the [Hub](https://huggingface.co/models). There are tags on the Hub that allow you to filter for a model you'd like to use for your task. Once you've picked an appropriate model, load it with the corresponding `AutoModelFor` and [`AutoTokenizer`] class. For example, load the [`AutoModelForCausalLM`] class for a causal language modeling task:
+[`pipeline_dataset`]
+[`pipeline_webserver`]
+
+## Parameters
+
+[`pipeline`] supports many parameters, some are task specific, some are general for all pipelines.
+In general you can specify parameters anywhere you want:
 
 ```py
->>> from transformers import AutoTokenizer, AutoModelForCausalLM
+>>> generator(model="openai/whisper-large", my_parameter=1)
+>>> out = generate(...) # This will use `my_parameter=1`.
+>>> out = generate(..., my_parameter=2) # This will override and use `my_parameter=2`.
+>>> out = generate(...) # This will go back to using `my_parameter=1`.
+```
+
+Let's check out 3 important ones:
 
->>> tokenizer = AutoTokenizer.from_pretrained("distilgpt2")
->>> model = AutoModelForCausalLM.from_pretrained("distilgpt2")
+### Device
+
+If you use `device=n` the pipeline will automatically put the model on said device.
+This will work regardless wether you are using `PyTorch` or `Tensorflow`.
+
+If the model is too large for a single GPU, you can also use [`accelerate`] (`device_map="auto"`).
+# TODO check link.
+
+```py
+>>> generator(model="openai/whisper-large", device=0)
 ```
 
-Create a [`pipeline`] for your task, and specify the model and tokenizer you've loaded:
+### Batch size
+
+By default, pipelines will not batch inference for reasons explained in detail [here](https://huggingface.co/docs/transformers/main_classes/pipelines#pipeline-batching). The reason is that batching is not necessarily faster, and can actually be quite slower in some cases.
+
+But if it works in your use case, you can use:
 
 ```py
->>> from transformers import pipeline
+>>> generator(model="openai/whisper-large", device=0, batch_size=2)
```

Comment:
Same here, unless the doctests are run on a machine with GPU @ydshieh ?

Reply:
doctest CI is run on a GPU machine. But it might be easier to make it work on CPU too - so the team members can debug more easily whenever something have to be fixed.

(sometimes it's easier/quicker to access a CPU machine - although I think everyone have GPU machines)

Reply:
I still removed it, not sure there's a lot of value added  in checking those values (it's tested against in the CI by fast tests anyway for all pipelines)